mkt/api/tests/test_middleware.py

```diff
@@ -333,3 +334,12 @@ def test_settings(self):
         index = settings.MIDDLEWARE_CLASSES.index
 
         ok_(index(auth_middleware) > index(api_middleware))
+
+    @mock.patch('mkt.api.middleware.log')
+    def test_shared_secret_no_break_restoauth(self, mock_log):
+        shared_secret = 'mkt-shared-secret me@email.com,hash'
+        request = RequestFactory(HTTP_AUTHORIZATION=shared_secret).get('/')
+        request.API = True
+        RestSharedSecretMiddleware().process_request(request)
```

Comment:
Since you are no longer popping the header you don't need to call `RestSharedSecretMiddleware`, do you ?